### PR TITLE
make modals display in their corresponding window

### PR DESCRIPTION
**Pull request by BurningTreeC**
this makes modals display in the window where they were triggered

### CODE REVIEW

File: core/modules/utils/dom/modal.js [hunk 1]
```diff
@@ -26,8 +26,9 @@ Display a modal dialogue
 Options include:
 	downloadLink: Text of a big download link to include
 */
-Modal.prototype.display = function(title,options) {
+Modal.prototype.display = function(title,options,currentdocument) {
 	options = options || {};
+	currentdocument = currentdocument || document;
```
Comment (pmario) [inline core/modules/utils/dom/modal.js]: IMO should be `currentDocument` if possible from the API

**Comment by Jermolene** [comment]
Hi @BurningTreeC good catch -- it's great that you're finding so many wrinkles with the multiwindow support. It was added fairly recently; perhaps we should have audited usages of document and window and been more methodical about tracking down problems.

I think @pmario is right that `currentDocument` would be preferred.

**Comment by BurningTreeC** [comment]
@Jermolene - thanks, I'd still like to find some more ... anyway, I've changed the name to "currentDocument"

I understand that parts of the core are designed without multi-window functionality in mind but I want to tackle what I think I understand. The storyviews are tricky, though.

I've made [another PR](#3332) recently for cancelling popups in new windows... would you mind having a look at it to see if it's any good if you find the time?

**Comment by Jermolene** [comment]
Hi @BurningTreeC I hope everything is OK; I see you've been closing quite a few PRs, is that intentional? I can understand if you've grown frustrated with the delays with merging, but would like to resolve things more constructively if possible.

**Pull request closed without merging**